sstables/mx/reader.cc

```diff
@@ -903,7 +902,7 @@ class data_consume_rows_context_m : public data_consumer::continuous_data_consum
         , _header(sst->get_serialization_header())
         , _column_translation(sst->get_column_translation(s, _header, sst->features()))
         , _has_shadowable_tombstones(sst->has_shadowable_tombstones())
-        , _gen(do_process_state(false))
+        , _gen(do_process_state())
```

Comment:
The commit message says just "remove skip_partition". Why? Can you write a few additional words on why it was needed in the past and no longer is?

Reply:
This will also appear only in the final version in the PR (the skip_partition was also added by me in an earlier commit in this series). The skip_partition was useful for checking if we called reset() and changed the state with that, but this can be done just by checking the state.